packages/run-protocol/src/psm/psm.js

```diff
@@ -0,0 +1,187 @@
+    const maxAnchor = floorMultiplyBy(afterFee, anchorPerStable);
+    // TODO this prevents the reallocate from failing. Can this be tested otherwise?
+    assert(
+      AmountMath.isGTE(maxAnchor, wanted),
+      X`wanted ${wanted} is more then ${given} minus fees ${fee}`,
+    );
+    stageTransfer(seat, stage, { In: afterFee }, { Stable: afterFee });
+    stageTransfer(seat, feePool, { In: fee }, { Stable: fee });
+    stageTransfer(anchorPool, seat, { Anchor: maxAnchor }, { Out: maxAnchor });
+    zcf.reallocate(seat, anchorPool, stage, feePool);
+    stableMint.burnLosses({ Stable: afterFee }, stage);
+  };
+
+  const wantStable = (seat, given, wanted = emptyStable) => {
+    const anchorAfterTrade = AmountMath.add(
+      anchorPool.getAmountAllocated('Anchor', anchorBrand),
+      given,
+    );
+    assertUnderLimit(anchorAfterTrade);
+    const asStable = floorDivideBy(given, anchorPerStable);
+    const fee = ceilMultiplyBy(asStable, gov.getWantStableRate());
+    const afterFee = AmountMath.subtract(asStable, fee);
+    assert(
+      AmountMath.isGTE(afterFee, wanted),
+      X`wanted ${wanted} is more then ${given} minus fees ${fee}`,
+    );
+    stableMint.mintGains({ Stable: asStable }, stage);
+
+    stageTransfer(seat, anchorPool, { In: given }, { Anchor: given });
+    stageTransfer(stage, seat, { Stable: afterFee }, { Out: afterFee });
+    stageTransfer(stage, feePool, { Stable: fee });
+    try {
+      zcf.reallocate(seat, anchorPool, stage, feePool);
+    } catch (e) {
+      // TODO reallocate should guarantee that this case cannot happen
+      stableMint.burnLosses({ Stable: asStable }, stage);
+      throw e;
+    }
+  };
+
+  const swapHook = seat => {
+    assertProposalShape(seat, {
+      give: { In: null },
+    });
+    const {
+      give: { In: given },
+      want: { Out: wanted } = { Out: undefined },
+    } = seat.getProposal();
+    if (given.brand === stableBrand) {
+      giveStable(seat, given, wanted);
+    } else if (given.brand === anchorBrand) {
+      wantStable(seat, given, wanted);
+    } else {
+      throw Error(`unexpected brand ${given.brand}`);
+    }
+    seat.exit();
+  };
+  const makeSwapInvitation = () => zcf.makeInvitation(swapHook, 'swap');
+
+  // Eventually the reward pool will live elsewhere. For now it's here for
+  // bookkeeping. It's needed in tests.
+  const getRewardAllocation = () => feePool.getCurrentAllocation();
+
+  const publicFacet = Far('Parity Stability Module', {
+    getCurrentLiquidity: () =>
+      anchorPool.getAmountAllocated('Anchor', anchorBrand),
+    makeSwapInvitation,
+    // getContractGovernor: () => electionManager,
+  });
+
+  const { makeCollectFeesInvitation } = makeMakeCollectFeesInvitation(
+    zcf,
+    feePool,
+    stableBrand,
+    'Stable',
+  );
+  const creatorFacet = Far('Parity Stability Module', {
+    getRewardAllocation,
+    makeCollectFeesInvitation,
+    // getContractGovernor: () => electionManager,
```

Comment:
stray. if this is meant to be left for future work, leave a comment. (though I think it's better to remove)